cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -629,7 +636,8 @@ private V1ObjectMeta createResourceMetadata(Type resourceType, String runnableNa
       .addToLabels(podInfo.getContainerLabelName(), runnableName)
       .addToLabels(KubeMasterEnvironment.NAMESPACE_PROPERTY, programRuntimeNamespace)
       .addToAnnotations(KubeTwillRunnerService.APP_LABEL, twillSpec.getName())
-      .addToAnnotations(KubeTwillRunnerService.START_TIMEOUT_ANNOTATION, Long.toString(startTimeoutMillis));
+      .addToAnnotations(KubeTwillRunnerService.START_TIMEOUT_ANNOTATION, Long.toString(startTimeoutMillis))
+      .addToAnnotations(kubeTwillConfigs);
```

Comment:
Why adding all configs to annotation? Generally not a good idea since from this class perspective, it doesn't check what are inside the configs map, which potentially contains things that shouldn't go into the annotation.

Reply:
makes sense, I will explicitly add only my config in the annotation here.